### PR TITLE
Accessibility: Add BoundingRects to UiaTextRanges

### LINKED ISSUE #2103: Feature Request: Fix Accessibility BoundingRects for TextRanges
# Summary of the new feature/enhancement

Thanks to #2083, we now have BoundingRects for text ranges. We can modify them and explore the text buffer properly.

Just one little problem...
image

The BoundingRects aren't actually on the text ranges properly. They update and function correctly, but they should look something like how they do in ConHost:
image

# Proposed technical implementation details (optional)
There's a few places in UiaTextRange that use `ScreenToClient()` and `ClientToScreen();` We can use them in ConHost because we have an HWND. Unfortunately, Windows Terminal doesn't have access to an HWND (and it seems like that'll be really hard to do).

XAML, on the other hand, should already know some information about bounding rects. That's how we got the one around the TermControl to appear and all of the other XAML elements.

We need to investigate if/how we can leverage XAML's knowledge of screen bounding rects and just pass them along to the UiaTextRange.

### PR DESCRIPTION
**Pull request by carlos-zamora**
## Summary of the Pull Request
Add the visible rectangle indicators to text ranges. They are now sized appropriately and are located in their correct position. This supports panes, and makes them fairly accessible to use.

 
## References
🚨 Target branch is "Refactor Accessibility Providers" #2414

## PR Checklist
* [x] Closes #2103 
* [x] CLA signed.
* [x] ~Tests added/passed~
* [x] ~Requires documentation to be updated~
* [x] I am a core contributor.

## Detailed Description of the Pull Request / Additional comments
Thanks to that refactoring, this code is fairly easy to read. There are a few additional refactoring changes throughout this commit.

The core of these changes are in `UiaTextRange::_TranslatePointToScreen()`. This is where we translate the RECTs to the correct size and position.
- Size: we needed the current font size. Acquired through `ScreenInfoUiaProvider` --> `TermControl`
  - 🚨 NOTE: using `IRenderData`'s is incorrect because Terminal has a fake one there.
- Position:
  - using `ScreenInfoUiaProvider`'s `get_BoundingRectangle` to leverage XAML's knowledge of where we are on the screen
    - XAML already uses this function to create a box around the TerminalControl. Let's just borrow that.
  - using `TermControl`'s `padding` value to adjust even further.

## Validation Steps Performed
1. Create various panes
2. Make one of the panes have default font size, another with larger, and another with smaller
3. For each pane, use inspect.exe to test as follows:
   1. get document range
   2. do bounding rects align properly with the text ✔
   3. create a selection
   4. get selection range
   5. do bounding rects align properly with the selection ✔
4. verify that each pane creates bounding rects of different sizes for the text.

### CODE REVIEW

File: src/cascadia/TerminalControl/ScreenInfoUiaProvider.cpp [hunk 1]
```diff
@@ -77,10 +82,20 @@ IFACEMETHODIMP ScreenInfoUiaProvider::get_FragmentRoot(_COM_Outptr_result_mayben
     return S_OK;
 }
 
-std::deque<Microsoft::Console::Types::UiaTextRangeBase*> ScreenInfoUiaProvider::GetSelectionRangeUTRs(_In_ Microsoft::Console::Render::IRenderData* pData,
-                                                                                                      _In_ IRawElementProviderSimple* pProvider)
+const COORD ScreenInfoUiaProvider::getFontSize() const
```
Comment (DHowett-MSFT) [inline src/cascadia/TerminalControl/ScreenInfoUiaProvider.cpp]: nit: `getFontSize` shouldn't have a lowercase `g`

File: src/cascadia/TerminalControl/UiaTextRange.cpp [hunk 2]
```diff
@@ -213,12 +213,39 @@ void UiaTextRange::_ChangeViewport(const SMALL_RECT NewWindow)
...
+    // update based on TermControl location (important for Panes)
+    UiaRect boundingRect;
+    THROW_IF_FAILED(provider->get_BoundingRectangle(&boundingRect));
+    clientPoint->x += gsl::narrow<LONG>(boundingRect.left);
+    clientPoint->y += gsl::narrow<LONG>(boundingRect.top);
+
+    // update based on TermControl padding
+    auto padding = provider->GetPadding();
+    clientPoint->x += gsl::narrow<LONG>(padding.Left);
+    clientPoint->y += gsl::narrow<LONG>(padding.Right);
```
Comment (DHowett-MSFT) [inline src/cascadia/TerminalControl/UiaTextRange.cpp]: You probably want `Top` :wink:

File: src/cascadia/TerminalControl/ScreenInfoUiaProvider.cpp [hunk 3]
```diff
@@ -77,10 +82,20 @@ IFACEMETHODIMP ScreenInfoUiaProvider::get_FragmentRoot(_COM_Outptr_result_mayben
...
+{
+    return _termControl.GetActualFont().GetSize();
+}
+
+const winrt::Windows::UI::Xaml::Thickness ScreenInfoUiaProvider::GetPadding() const
+{
+    return _termControl.GetPadding();
+}
+
+std::deque<UiaTextRangeBase*> ScreenInfoUiaProvider::GetSelectionRangeUTRs(_In_ IRenderData* pData,
```
Comment (adiviness) [inline src/cascadia/TerminalControl/ScreenInfoUiaProvider.cpp]: missing consts on params, here and a bunch below

File: src/cascadia/TerminalControl/ScreenInfoUiaProvider.cpp [hunk 4]
```diff
@@ -92,31 +107,31 @@ std::deque<Microsoft::Console::Types::UiaTextRangeBase*> ScreenInfoUiaProvider::
     return result;
 }
 
-Microsoft::Console::Types::UiaTextRangeBase* ScreenInfoUiaProvider::CreateUTR(_In_ Microsoft::Console::Render::IRenderData* pData,
-                                                                              _In_ IRawElementProviderSimple* const pProvider)
+UiaTextRangeBase* ScreenInfoUiaProvider::CreateUTR(_In_ IRenderData* pData,
```
Comment (adiviness) [inline src/cascadia/TerminalControl/ScreenInfoUiaProvider.cpp]: Microsoft coding style is to capitalize only the first letter of an acronym.

File: src/cascadia/TerminalControl/ScreenInfoUiaProvider.cpp (hunk 3 above)
Comment (carlos-zamora) [reply src/cascadia/TerminalControl/ScreenInfoUiaProvider.cpp]: :D you're back, Still catching me for that `const` thing haha

**Comment by carlos-zamora** [comment]
Rebased off of the updated base branch.
Heads up, I'll be sure this goes into master as a separate commit from the base branch **after** the base branch goes into master. Just did this for the better diff :)

**Comment by carlos-zamora** [comment]
> I approve of this PR, but I'm not totally sure that GH will let you just change the target branch of a PR. You might need to make another PR once #2414 is in master. Either way, count this as my signoff on this diff

I know if you click the "Edit" button at the top, you can retarget. _Hopefully_, the approvals transfer over. But this'll be a learning experience for all of us #DayOfLearning haha

**Pull request merged**